crates/turborepo-filewatch/src/hash_watcher.rs

```diff
@@ -253,21 +265,48 @@ impl FileHashes {
         }
     }
 
-    fn get_package_path(&self, file_path: &AnchoredSystemPath) -> Option<&AnchoredSystemPath> {
+    fn get_changed_specs(&self, file_path: &AnchoredSystemPath) -> HashSet<HashSpec> {
         self.0
             .get_ancestor(file_path.as_str())
-            .and_then(|subtrie| subtrie.key())
-            .map(|package_path| {
-                AnchoredSystemPath::new(package_path).expect("keys are valid AnchoredSystemPaths")
-            })
-            .filter(|package_path| {
+            // verify we have a key
+            .and_then(|subtrie| subtrie.key().map(|key| (key, subtrie)))
+            // convert key to AnchoredSystemPath, and verify we have a value
+            .and_then(|(package_path, subtrie)| {
+                let package_path = AnchoredSystemPath::new(package_path)
+                    .expect("keys are valid AnchoredSystemPaths");
                 // handle scenarios where even though we've found an ancestor, it might be a
                 // sibling file or directory that starts with the same prefix,
                 // e,g an update to apps/foo_decoy when the package path is
-                // apps/foo. Note that relation_to_path will return Parent for
-                // equivalent paths.
-                package_path.relation_to_path(file_path) == PathRelation::Parent
+                // apps/foo.
+                if let Some(package_path_to_file) = file_path.strip_prefix(package_path) {
+                    // Pass along the path to the package, the path _within_ the package to this
+                    // change, in unix format, and the set of input specs that
+                    // we're tracking.
+                    subtrie
+                        .value()
+                        .map(|specs| (package_path, package_path_to_file.to_unix(), specs))
+                } else {
+                    None
+                }
```

Comment:
Can flatten this a bit with `?`
```suggestion
                let package_path_to_file = file_path.strip_prefix(package_path)?;
                // Pass along the path to the package, the path _within_ the package to this
                // change, in unix format, and the set of input specs that
                // we're tracking.
                let specs = subtrie.value()?;
                Some((package_path, package_path_to_file.to_unix(), specs))
```